source/jobs/README.md

```diff
@@ -0,0 +1,232 @@
+a request to receive the latest pending jobs. It will use the shared MQTT connection to subscribe/publish to these
+topics. 
+
+When a new Job is received via these topics, the Jobs feature will look for and extract the following required and optional elements:
+
+
+### Creating a Job
+
+[View a sample job document here](sample-job-docs/install-packages.json)
+
+The following fields are extracted from the incoming Job document by the Jobs feature within the AWS IoT Device Client when a 
+job is received.
+
+`operation` *string* (Required): This is the executable or script that you'd like to run. Here we can specify a script located
+in the Device Client's handler directory, such as `install-packages.sh`, or it could be a well known executable binary
+in the Device Client's environment path, such as `echo` or `apt-get`.
+
+`args` *string array* (Optional): This is a JSON array of string arguments which will be passed to the specified operation. For example, 
+if we want to run our `install-packages.sh` to install the package `ifupdown`, our args field would look like:
+```
+...
+"args": ["ifupdown"],
+...
+```
+If we wanted our remote device to print out "Hello World", then our job document so far might look like:
+```
+{
+    "operation": "echo",
+    "args": ["Hello world!"]
+}
+```
+
+`path` *string* (Optional): This field tells the Jobs feature where it should look to find an executable that matches the specified
+operation. If this field is omitted, the Jobs feature will assume that the executable can be found in its path. If
+you expect that the executable (operation) should be found in the Job feature's handler directory 
+(~/.aws-iot-device-client/jobs by default or specified via command line/json configuration values), then the path
+must be specified as part of the job document as follows:
+```
+...
+"path": "default",
+...
+```
+The path could also be specified as a particular directory that is expected on the device, such as:
+```
+...
+"path": "/home/ubuntu/my-job-handler-folder"
+...
+```
+
+`includeStdOut` *boolean* (Optional): This field tells the Jobs feature whether the standard output (STDOUT) of the child process
+should be included in the job execution update that is published to the IoT Jobs service. For example:
+```
+...
+"includeStdOut": true,
+...
+```
+
+`allowStdErr` *integer* (Optional): This field tells the Jobs feature whether a specific number of lines of output on STDERR
+from the job handler may be acceptable. If this field is omitted, the Jobs feature will use a default of 0 (zero), meaning
+that any lines of output on STDERR will cause the job to be marked as failed. For example: 
+```
+...
+"allowStdErr": 2,
+...
+```
+This job document indicates that the Jobs feature will mark the job as a success if the return code of the process is 0 AND
+there are fewer than or equal to 2 (two) lines of standard error output from the job handler. 
+
+### Creating your own Job Handler
```

Comment:
Should we also mention that we provide some handlers by default and list/link them?